app/settings/developer-apps/-components/app-form/component.ts

```diff
@@ -0,0 +1,93 @@
+import { tagName } from '@ember-decorators/component';
+import Component from '@ember/component';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+import RouterService from '@ember/routing/router-service';
+import { inject as service } from '@ember/service';
+import { ChangesetDef } from 'ember-changeset/types';
+import { TaskInstance } from 'ember-concurrency';
+import { task } from 'ember-concurrency-decorators';
+import Intl from 'ember-intl/services/intl';
+import Toast from 'ember-toastr/services/toast';
+
+import DeveloperApp, { developerAppValidations } from 'ember-osf-web/models/developer-app';
+import buildChangeset from 'ember-osf-web/utils/build-changeset';
+import captureException, { getApiErrorMessage } from 'ember-osf-web/utils/capture-exception';
+
+@tagName('')
+export default class DeveloperAppForm extends Component {
+    @service intl!: Intl;
+    @service router!: RouterService;
+    @service toast!: Toast;
+
+    // Required arguments
+    developerApp?: DeveloperApp;
+    appTaskInstance?: TaskInstance<DeveloperApp>;
+    mode!: string;
```

Comment:
I don't really like this string method of toggling what is shown on the template, so any suggestions here would be welcome

Reply:
If it's just toggling between create mode or not, you could make a createMode boolean to set if `mode == 'create'`, and use that for the template logic.

Reply:
It seems so obvious now that is should just be a boolean...